src/Symfony/Component/Process/Process.php

```diff
@@ -1237,14 +1237,31 @@ public static function isPtySupported()
             return $result;
         }
 
-        if ('\\' === DIRECTORY_SEPARATOR) {
+        if (self::isWindowsPlatform()) {
             return $result = false;
         }
 
         return $result = (bool) @proc_open('echo 1', array(array('pty'), array('pty'), array('pty')), $pipes);
     }
 
     /**
+     * Returns whether the current process is run on Windows platform (Command Prompt, PowerShell) or not.
+     * 
+     * Keep in mind, that Windows users can use Unix shells (like the git-shell) with TTY/PTY support,
+     * in this case, this function will return 'false'.
+     *
+     * @return bool
+     */
+    public static function isWindowsPlatform()
```

Comment:
-1 for introducing ``isWindowsPlatform``. Inlining the constant comparison is fine (and it is what is already used today).

You cannot assume that using a Unix shell on windows is the same than running on Unix:
- this component handles a bunch of windows-only PHP bugs related to subprocesses, which will still be there if you use git bash (as this does not change your PHP)
- killing tasks on windows does not depend on which shell you use, but only on the OS

and regarding this PR, I'm not even sure it will make tty work on Windows when running from git bash. I don't think ``/dev/tty`` is actually accessible for subprocesses launched by PHP (these processes are not wrapped in git bash).